Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors.

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24379 Implement operator logical_and
* #23978 Let logical_xor support non-bool tensors.
* #23916 Let logical_not support non-bool tensors.
* **#23860 Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors.**
* #23847 Add logical_xor operator
* #23839 Add logical_not operator.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors.

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

gchanan reviewed (approved):
looks good to me.  Note to whoever is looking to land this: don't try to land before prior PRs in the stack are ready.

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

gchanan commented:
clang-tidy failure looks bogus.

xuhdev commented:
@pytorchbot merge this please

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Close #23836

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

gchanan commented:
@xuhdev I've been trying to land this, we are having some internal issues with our landing system though, hopefully it will be resolved soon.

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)

xuhdev committed:
Update on "Recommend logical_not() instead of bitwise_not() when applying sub and neg on bool tensors."


Pull Request resolved: https://github.com/pytorch/pytorch/pull/23860

Differential Revision: [D16678299](https://our.internmc.facebook.com/intern/diff/D16678299)
